core/src/main/java/com/linecorp/armeria/internal/common/Http2ObjectEncoder.java

```diff
@@ -82,11 +82,15 @@ public final ChannelFuture doWriteData(int id, int streamId, HttpData data, bool
     }
 
     @Override
-    public final ChannelFuture doWriteReset(int id, int streamId, Http2Error error) {
+    public final ChannelFuture doWriteReset(int id, int streamId, Http2Error error,
+                                            boolean sendResetIfRemoteIsOpen) {
         final Http2Stream stream = encoder.connection().stream(streamId);
+
         // Send a RST_STREAM frame only for an active stream which did not send a RST_STREAM frame already.
         if (stream != null && !stream.isResetSent()) {
-            return encoder.writeRstStream(ctx, streamId, error.code(), ctx.newPromise());
+            if (!sendResetIfRemoteIsOpen || stream.state().remoteSideOpen()) {
```

Comment:
Q: Do we need to send RST_FRAME when half-close on remote? An EOF for the stream has been received from the remote peer already.

Reply:
Change not to send RST if EOF is received. 😉